### Problem Summary:
The Challenge was divided into the following three parts, each of which had specific requirements that
pertained to identifying anomalous entities in different aspects of the Medicare system:

In [1]:
import numpy as np
import pandas as pd

In [2]:
from matplotlib import pyplot as plt
import seaborn as sns

#### Importing Inpatient Data for both Medicare and Provider

In [3]:
medicare_inpatient = pd.read_csv('Medicare_Charge_Inpatient_DRG100_DRG_Summary_by_DRG_FY2011.csv', sep=',')
medicare_inpatient.head()

DRG Definition  Total Discharges  \
0           039 - EXTRACRANIAL PROCEDURES W/O CC/MCC             37861   
1  057 - DEGENERATIVE NERVOUS SYSTEM DISORDERS W/...             37927   
2  064 - INTRACRANIAL HEMORRHAGE OR CEREBRAL INFA...             68163   
3  065 - INTRACRANIAL HEMORRHAGE OR CEREBRAL INFA...            110983   
4  066 - INTRACRANIAL HEMORRHAGE OR CEREBRAL INFA...             62346   

   Average Covered Charges  Average Total Payments  Average Medicare Payments  
0              29228.55368             6902.381289                5514.435910  
1              23924.98059             6809.799325                5791.920242  
2              49186.95644            13614.616130               12415.825360  
3              30371.27399             8037.269492                6858.689088  
4              23300.06199             5762.881821                4508.202659

In [4]:
provider_inpatient = pd.read_csv('Medicare_Provider_Charge_Inpatient_DRG100_FY2011.csv', sep=',')
provider_inpatient.head()

DRG Definition  Provider Id  \
0  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10001   
1  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10005   
2  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10006   
3  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10011   
4  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10016   

                      Provider Name     Provider Street Address Provider City  \
0  SOUTHEAST ALABAMA MEDICAL CENTER      1108 ROSS CLARK CIRCLE        DOTHAN   
1     MARSHALL MEDICAL CENTER SOUTH  2505 U S HIGHWAY 431 NORTH          BOAZ   
2    ELIZA COFFEE MEMORIAL HOSPITAL          205 MARENGO STREET      FLORENCE   
3                 ST VINCENT'S EAST  50 MEDICAL PARK EAST DRIVE    BIRMINGHAM   
4     SHELBY BAPTIST MEDICAL CENTER     1000 FIRST STREET NORTH     ALABASTER   

  Provider State  Provider Zip Code  \
0             AL              36301   
1             AL              35957   
2             AL              35631   
3             AL              35235   
4             AL              35007   

  Hospital Referral Region (HRR) Description  Total Discharges  \
0                                AL - Dothan                91   
1                            AL - Birmingham                14   
2                            AL - Birmingham                24   
3                            AL - Birmingham                25   
4                            AL - Birmingham                18   

   Average Covered Charges  Average Total Payments  Average Medicare Payments  
0              32963.07692             5777.241758                4763.736264  
1              15131.85714             5787.571429                4976.714286  
2              37560.37500             5434.958333                4453.791667  
3              13998.28000             5417.560000                4129.160000  
4              31633.27778             5658.333333                4851.444444

#### Importing Outpatient Data for both Medicare and Provider

In [5]:
medicare_outpatient = pd.read_csv('Medicare_Charge_Outpatient_APC30_Summary_by_APC_CY2011.csv', sep=',')
medicare_outpatient.head()

APC  Outpatient Services  \
0    0012 - Level I Debridement & Destruction               275987   
1   0013 - Level II Debridement & Destruction               434706   
2  0015 - Level III Debridement & Destruction               509576   
3             0019 - Level I Excision/ Biopsy                27860   
4            0020 - Level II Excision/ Biopsy                40642   

   Average Estimated Submitted Charges  Average Total Payments  
0                            74.202956               16.147079  
1                           196.170070               56.133853  
2                           326.503444               97.156250  
3                          2013.181972              330.908369  
4                          3161.457873              567.156154

In [6]:
provider_outpatient = pd.read_csv('Medicare_Provider_Charge_Outpatient_APC30_CY2011_v2.csv', sep=',')
provider_outpatient.head()

APC  Provider Id  \
0  0012 - Level I Debridement & Destruction        10029   
1  0012 - Level I Debridement & Destruction        20024   
2  0012 - Level I Debridement & Destruction        30064   
3  0012 - Level I Debridement & Destruction        30088   
4  0012 - Level I Debridement & Destruction        30111   

                                       Provider Name  \
0                EAST ALABAMA MEDICAL CENTER AND SNF   
1                 CENTRAL PENINSULA GENERAL HOSPITAL   
2   UNIVERSITY OF ARIZONA MEDICAL CTR-UNIVERSIT, THE   
3                      BANNER BAYWOOD MEDICAL CENTER   
4  UNIVERSITY OF ARIZONA MEDICAL CTR- SOUTH CAM, THE   

      Provider Street Address Provider City Provider State  Provider Zip Code  \
0      2000 PEPPERELL PARKWAY       OPELIKA             AL              36801   
1          250 HOSPITAL PLACE      SOLDOTNA             AK              99669   
2  1501 NORTH CAMPBELL AVENUE        TUCSON             AZ              85724   
3    6644 EAST BAYWOOD AVENUE          MESA             AZ              85206   
4           2800 EAST AJO WAY        TUCSON             AZ              85713   

  Hospital Referral Region (HRR) Description  Outpatient Services  \
0                            AL - Birmingham                   23   
1                             AK - Anchorage                  994   
2                                AZ - Tucson                 1765   
3                                  AZ - Mesa                   20   
4                                AZ - Tucson                   22   

   Average  Estimated Submitted Charges  Average Total Payments  
0                             78.086957               21.910435  
1                            149.589749               36.623853  
2                             50.135411               14.541841  
3                            112.400000               23.736000  
4                            152.045455               16.569091

In [7]:
# info about medicare_inpatient

medicare_inpatient.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 5 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   DRG Definition             100 non-null    object 
 1   Total Discharges           100 non-null    int64  
 2   Average Covered Charges    100 non-null    float64
 3   Average Total Payments     100 non-null    float64
 4   Average Medicare Payments  100 non-null    float64
dtypes: float64(3), int64(1), object(1)
memory usage: 4.0+ KB


In [8]:
# info about provider_inpatient

provider_inpatient.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 163065 entries, 0 to 163064
Data columns (total 12 columns):
 #   Column                                      Non-Null Count   Dtype  
---  ------                                      --------------   -----  
 0   DRG Definition                              163065 non-null  object 
 1   Provider Id                                 163065 non-null  int64  
 2   Provider Name                               163065 non-null  object 
 3   Provider Street Address                     163065 non-null  object 
 4   Provider City                               163065 non-null  object 
 5   Provider State                              163065 non-null  object 
 6   Provider Zip Code                           163065 non-null  int64  
 7   Hospital Referral Region (HRR) Description  163065 non-null  object 
 8   Total Discharges                            163065 non-null  int64  
 9   Average Covered Charges                     163065 non-null  float64
 

In [9]:
# info about medicare_outpatient

medicare_outpatient.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 4 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   APC                                  30 non-null     object 
 1   Outpatient Services                  30 non-null     int64  
 2   Average Estimated Submitted Charges  30 non-null     float64
 3   Average Total Payments               30 non-null     float64
dtypes: float64(2), int64(1), object(1)
memory usage: 1.1+ KB


In [10]:
# info about provider_outpatient

provider_outpatient.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 43372 entries, 0 to 43371
Data columns (total 11 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   APC                                         43372 non-null  object 
 1   Provider Id                                 43372 non-null  int64  
 2   Provider Name                               43372 non-null  object 
 3   Provider Street Address                     43372 non-null  object 
 4   Provider City                               43372 non-null  object 
 5   Provider State                              43372 non-null  object 
 6   Provider Zip Code                           43372 non-null  int64  
 7   Hospital Referral Region (HRR) Description  43372 non-null  object 
 8   Outpatient Services                         43372 non-null  int64  
 9   Average  Estimated Submitted Charges        43372 non-null  float64
 10  Average To

In [12]:
provider_inpatient['DRG Definition'].nunique()

100

From each of the above info() can see there aren't any null values in the data

-------

### Part 1: Identify providers that overcharge for certain procedures or regions where procedures are too expensive.


### Part 1A: Highest Cost Variation

To find out the highest cost variation compute the Coefficient of Variation because it is measure that helps to compare the variation of distributions (or variation in the variables) with different units relative to their means. It is calculated by dividing Standard Deviation by Mean.

In [11]:
# A) For Inpatients on DRG: 

# Since we are checking for cost variation, hence would take Average Covered Charges in consideration:

res1 = provider_inpatient.pivot_table(index=['DRG Definition'],values=['Average Covered Charges'], 
                                      aggfunc={np.mean, np.std}).round(2).reset_index()
res1.head()

DRG Definition Average Covered Charges  \
                                                                        mean   
0           039 - EXTRACRANIAL PROCEDURES W/O CC/MCC                30481.99   
1  057 - DEGENERATIVE NERVOUS SYSTEM DISORDERS W/...                24307.76   
2  064 - INTRACRANIAL HEMORRHAGE OR CEREBRAL INFA...                48552.61   
3  065 - INTRACRANIAL HEMORRHAGE OR CEREBRAL INFA...                29492.13   
4  066 - INTRACRANIAL HEMORRHAGE OR CEREBRAL INFA...                23592.93   

             
        std  
0  15941.25  
1  14085.29  
2  29593.22  
3  16732.40  
4  12606.45

In [12]:
res1['Coefficient_Of_Variation'] = res1['Average Covered Charges', 'std']/res1['Average Covered Charges', 'mean']

In [13]:
res1 = res1.sort_values(by= 'Coefficient_Of_Variation', ascending=False).reset_index(drop=True)
res1.head(3)

DRG Definition Average Covered Charges  \
                                                                        mean   
0                                    885 - PSYCHOSES                19929.46   
1  897 - ALCOHOL/DRUG ABUSE OR DEPENDENCE W/O REH...                17589.10   
2     917 - POISONING & TOXIC EFFECTS OF DRUGS W MCC                40345.54   

            Coefficient_Of_Variation  
        std                           
0  14305.27                 0.717795  
1  11150.91                 0.633967  
2  24737.87                 0.613150

The top 3 DRG for Inpatients with highest Cost Variation are:
    
    1) 885 - PSYCHOSES
    2) 897 - ALCOHOL/DRUG ABUSE OR DEPENDENCE W/O REHABILITATION THERAPY W/O MCC
    3) 917 - POISONING & TOXIC EFFECTS OF DRUGS W MCC

In [14]:
## B) For Outpatients:

# Since we are checking for cost variation, hence would take Average Estimated Submitted Charges in consideration:

res2 = provider_outpatient.pivot_table(index=['APC'], values=['Average  Estimated Submitted Charges'],
                                      aggfunc={np.mean, np.std}).round(2).reset_index()

In [15]:
res2['Coefficient_of_Variation'] = res2['Average  Estimated Submitted Charges', 'std']/res2['Average  Estimated Submitted Charges', 'mean']

In [16]:
res2 = res2.sort_values(by='Coefficient_of_Variation', ascending=False).reset_index(drop=True)
res2.head(3)

APC  \
                                            
0   0604 - Level 1 Hospital Clinic Visits   
1  0698 - Level II Eye Tests & Treatments   
2         0019 - Level I Excision/ Biopsy   

  Average  Estimated Submitted Charges          Coefficient_of_Variation  
                                  mean      std                           
0                               169.78   235.18                 1.385204  
1                               250.68   328.50                 1.310436  
2                              2264.43  2785.01                 1.229894

The top 3 APC for Outpatients with highest Cost Variation are:
    
    1) 0604 - Level 1 Hospital Clinic Visits
    2) 0698 - Level II Eye Tests & Treatments
    3) 0019 - Level I Excision/ Biopsy

------------------------------------------------------

### Parts 1B Highest-Cost Claims by Provider

In [40]:
# A) For Inpatients on DRG:
g1 = provider_inpatient[['DRG Definition','Provider Id','Provider Name','Average Medicare Payments']]
g1.head()

DRG Definition  Provider Id  \
0  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10001   
1  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10005   
2  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10006   
3  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10011   
4  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10016   

                      Provider Name  Average Medicare Payments  
0  SOUTHEAST ALABAMA MEDICAL CENTER                4763.736264  
1     MARSHALL MEDICAL CENTER SOUTH                4976.714286  
2    ELIZA COFFEE MEMORIAL HOSPITAL                4453.791667  
3                 ST VINCENT'S EAST                4129.160000  
4     SHELBY BAPTIST MEDICAL CENTER                4851.444444

In [47]:
res2 = g1.loc[g1.groupby(['DRG Definition'])['Average Medicare Payments'].idxmax()].sort_values(by= 'Average Medicare Payments',ascending=False).reset_index(drop=True)
res2.head()

DRG Definition  Provider Id  \
0  207 - RESPIRATORY SYSTEM DIAGNOSIS W VENTILATO...        50441   
1  853 - INFECTIOUS & PARASITIC DISEASES W O.R. P...       330234   
2        460 - SPINAL FUSION EXCEPT CERVICAL W/O MCC       150166   
3   870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS        50441   
4   329 - MAJOR SMALL & LARGE BOWEL PROCEDURES W MCC       330234   

                Provider Name  Average Medicare Payments  
0           STANFORD HOSPITAL                154620.8182  
1  WESTCHESTER MEDICAL CENTER                133177.2667  
2           PINNACLE HOSPITAL                130466.5714  
3           STANFORD HOSPITAL                109303.2143  
4  WESTCHESTER MEDICAL CENTER                100040.2667

The top 3 Providers that charge the highest-Cost Claims for Inpatients are:

1. Stanford Hospital
2. Westchester Medical Center
3. Pinnacle Hospital

In [52]:
# B) For Outpatients on APC:

g2 = provider_outpatient[['APC','Provider Id', 'Provider Name','Average Total Payments']]
g2.head()

APC  Provider Id  \
0  0012 - Level I Debridement & Destruction        10029   
1  0012 - Level I Debridement & Destruction        20024   
2  0012 - Level I Debridement & Destruction        30064   
3  0012 - Level I Debridement & Destruction        30088   
4  0012 - Level I Debridement & Destruction        30111   

                                       Provider Name  Average Total Payments  
0                EAST ALABAMA MEDICAL CENTER AND SNF               21.910435  
1                 CENTRAL PENINSULA GENERAL HOSPITAL               36.623853  
2   UNIVERSITY OF ARIZONA MEDICAL CTR-UNIVERSIT, THE               14.541841  
3                      BANNER BAYWOOD MEDICAL CENTER               23.736000  
4  UNIVERSITY OF ARIZONA MEDICAL CTR- SOUTH CAM, THE               16.569091

In [57]:
res2 = g2.loc[g2.groupby(['APC', 'Provider Id', 'Provider Name'])['Average Total Payments'].idxmax()].sort_values(by='Average Total Payments',ascending=False).reset_index(drop=True)
res2.head()

APC  Provider Id  \
0  0074 - Level IV Endoscopy Upper Airway        50441   
1  0074 - Level IV Endoscopy Upper Airway        50454   
2  0074 - Level IV Endoscopy Upper Airway        50764   
3  0074 - Level IV Endoscopy Upper Airway        50599   
4  0074 - Level IV Endoscopy Upper Airway       330195   

                                   Provider Name  Average Total Payments  
0                              STANFORD HOSPITAL             2062.470526  
1                            UCSF MEDICAL CENTER             1996.830000  
2                 SHASTA REGIONAL MEDICAL CENTER             1833.500000  
3  UNIVERSITY OF CALIFORNIA DAVIS MEDICAL CENTER             1805.955172  
4              LONG ISLAND JEWISH MEDICAL CENTER             1794.600000

The top 3 Providers that charge the highest-Cost Claims for Outpatients are:

1. Stanford Hospital
2. UCSF Medical Center
3. Shasta Regional Medical Center

------------------------------------------------------

### Parts 1C: Highest-Cost Claims by Region

In [60]:
# For Inpatients:
g3 = provider_inpatient[['DRG Definition','Hospital Referral Region (HRR) Description','Average Medicare Payments']] 
g3.head()

DRG Definition  \
0  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC   
1  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC   
2  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC   
3  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC   
4  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC   

  Hospital Referral Region (HRR) Description  Average Medicare Payments  
0                                AL - Dothan                4763.736264  
1                            AL - Birmingham                4976.714286  
2                            AL - Birmingham                4453.791667  
3                            AL - Birmingham                4129.160000  
4                            AL - Birmingham                4851.444444

In [63]:
res4 = g3.loc[g3.groupby(['DRG Definition'])['Average Medicare Payments'].idxmax()].sort_values(by='Average Medicare Payments', ascending=False).reset_index(drop=True)
res4.head()      

DRG Definition  \
0  207 - RESPIRATORY SYSTEM DIAGNOSIS W VENTILATO...   
1  853 - INFECTIOUS & PARASITIC DISEASES W O.R. P...   
2        460 - SPINAL FUSION EXCEPT CERVICAL W/O MCC   
3   870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS   
4   329 - MAJOR SMALL & LARGE BOWEL PROCEDURES W MCC   

  Hospital Referral Region (HRR) Description  Average Medicare Payments  
0                      CA - San Mateo County                154620.8182  
1                          NY - White Plains                133177.2667  
2                                  IN - Gary                130466.5714  
3                      CA - San Mateo County                109303.2143  
4                          NY - White Plains                100040.2667

The top 3 Regions that charge the highest-Cost Claims for Inpatients are:

1. CA - San Mateo County
2. NY - White Plains
3. IN - Gary

In [65]:
# For Outpatients:
g4 = provider_outpatient[['APC','Hospital Referral Region (HRR) Description', 'Average Total Payments']]
g4.head()


APC  \
0  0012 - Level I Debridement & Destruction   
1  0012 - Level I Debridement & Destruction   
2  0012 - Level I Debridement & Destruction   
3  0012 - Level I Debridement & Destruction   
4  0012 - Level I Debridement & Destruction   

  Hospital Referral Region (HRR) Description  Average Total Payments  
0                            AL - Birmingham               21.910435  
1                             AK - Anchorage               36.623853  
2                                AZ - Tucson               14.541841  
3                                  AZ - Mesa               23.736000  
4                                AZ - Tucson               16.569091

In [70]:
res5 = g4.loc[g4.groupby(['APC'])['Average Total Payments'].idxmax()].sort_values(by='Average Total Payments', ascending=False).reset_index(drop=True)
res5.head(10)

APC  \
0             0074 - Level IV Endoscopy Upper Airway   
1                   0203 - Level IV Nerve Injections   
2  0209 - Level II Extended EEG, Sleep, and Cardi...   
3                    0377 - Level II Cardiac Imaging   
4                   0020 - Level II Excision/ Biopsy   
5   0270 - Level III Echocardiogram Without Contrast   
6                  0207 - Level III Nerve Injections   
7  0336 - Magnetic Resonance Imaging and Magnetic...   
8    0269 - Level II Echocardiogram Without Contrast   
9                    0019 - Level I Excision/ Biopsy   

  Hospital Referral Region (HRR) Description  Average Total Payments  
0                      CA - San Mateo County             2062.470526  
1                      CA - San Mateo County             1218.580000  
2                      CA - San Mateo County             1072.982991  
3                              CA - San Jose             1050.740000  
4                              CA - San Jose              808.850000  
5                            CA - Santa Cruz              777.310000  
6                              CA - San Jose              712.091765  
7                             PA - Allentown              680.095556  
8                              CA - San Jose              556.400000  
9                        CA - Alameda County              470.720000

The top 3 Regions that charge the highest-Cost Claims for Inpatients are:

1. CA - San Mateo County
2. CA - San Jose
3. CA - Santa Cruz

------------------------------------------------------

### Part 1D: Highest Number of Procedures and Largest Differences between Claims and Reimbursements

In [84]:
# For Inpatients:
provider_inpatient['DRG Definition'].max()

'948 - SIGNS & SYMPTOMS W/O MCC'

In [80]:
# To find the largest difference b/w Claims and Reimbursements

provider_inpatient['diff'] = provider_inpatient['Average Total Payments'] - provider_inpatient['Average Medicare Payments']
provider_inpatient.max()

DRG Definition                                            948 - SIGNS & SYMPTOMS W/O MCC
Provider Id                                                                       670077
Provider Name                                 ZUNI COMPREHENSIVE COMMUNITY HEALTH CENTER
Provider Street Address                                          WILKENS & CATON AVENUES
Provider City                                                                       ZUNI
Provider State                                                                        WY
Provider Zip Code                                                                  99835
Hospital Referral Region (HRR) Description                                   WY - Casper
Total Discharges                                                                    3383
Average Covered Charges                                                           929119
Average Total Payments                                                            156158
Average Medicare Paym

In [82]:
# For Outpatients:
provider_outpatient['APC'].max()

'0698 - Level II Eye Tests & Treatments'

The highest number of procedures is 948 - SIGNS & SYMPTOMS W/O MCC and this procedure also has the largest difference between Claims and Reimbursements for Inpatients.

For Outpatients, the highest Procedure APC is 0698 - Level II Eye Tests & Treatments and don't have the largest difference between the claims and reimbursements as the no reimbursements are made for outpatients.

------------------------------------------------------

### Part 2: Identify the three providers that are least similar to other providers and the three regions that are least similar to other regions

### Part 2A: Providers Least Like Others

In [11]:
# For Inpatients:

res7 = provider_inpatient.groupby(['Provider Id','Provider Name'])[['Total Discharges','Average Covered Charges']].mean().round(2).add_prefix('Avg_Of_').reset_index()
res7

Provider Id                           Provider Name  \
0           10001        SOUTHEAST ALABAMA MEDICAL CENTER   
1           10005           MARSHALL MEDICAL CENTER SOUTH   
2           10006          ELIZA COFFEE MEMORIAL HOSPITAL   
3           10007                MIZELL MEMORIAL HOSPITAL   
4           10008             CRENSHAW COMMUNITY HOSPITAL   
...           ...                                     ...   
3332       670072        WESTBURY COMMUNITY HOSPITAL, LLC   
3333       670073          METHODIST HOSPITAL FOR SURGERY   
3334       670075  ST LUKE'S HOSPITAL AT THE VINTAGE, LLC   
3335       670076         HERITAGE PARK SURGICAL HOSPITAL   
3336       670077         METHODIST WEST HOUSTON HOSPITAL   

      Avg_Of_Total Discharges  Avg_Of_Average Covered Charges  
0                       58.75                        35247.03  
1                       28.96                        16451.10  
2                       45.36                        36942.36  
3                       27.41                        12079.54  
4                       17.89                        16148.76  
...                       ...                             ...  
3332                   553.00                         9132.27  
3333                    54.75                        26091.04  
3334                    13.50                        33840.39  
3335                    18.00                        33310.61  
3336                    14.00                        22316.36  

[3337 rows x 4 columns]

In [15]:
res7.sort_values(by=['Avg_Of_Total Discharges','Avg_Of_Average Covered Charges'], ascending=[False,False]).head(10)

Provider Id                               Provider Name  \
1444       220062            ADCARE HOSPITAL OF WORCESTER INC   
3332       670072            WESTBURY COMMUNITY HOSPITAL, LLC   
2058       330270                HOSPITAL FOR SPECIAL SURGERY   
2294       360247                       WOODS AT PARKSIDE,THE   
206         40147                  ARKANSAS SURGICAL HOSPITAL   
3285       520205  MIDWEST ORTHOPEDIC SPECIALTY HOSPITAL, LLC   
1189       170188           KANSAS CITY ORTHOPAEDIC INSTITUTE   
599        100007                            FLORIDA HOSPITAL   
2298       360266   MOUNT CARMEL NEW ALBANY SURGICAL HOSPITAL   
1185       170183            KANSAS SURGERY & RECOVERY CENTER   

      Avg_Of_Total Discharges  Avg_Of_Average Covered Charges  
1444                  1571.00                         4227.47  
3332                   553.00                         9132.27  
2058                   494.50                        57047.78  
2294                   431.00                         5384.69  
206                    306.75                        25321.48  
3285                   293.00                        41004.60  
1189                   260.00                        29787.77  
599                    258.28                        51151.35  
2298                   250.00                        33315.32  
1185                   245.50                        23256.21

On the basis of the highest number of average total discharges and average covered Charges, the 3 Providers that are least like others are amongst the Inpatients:

1. Adcare Hospital of Worcester Inc
2. Westbury Community Hospital, LLC
3. Hospital For Special Surgery

As these providers stand as outliers.

In [37]:
# For Outpatients:
res8 = provider_outpatient.groupby(['Provider Id', 'Provider Name'])[['Outpatient Services','Average  Estimated Submitted Charges']].mean().round(2).add_prefix('Avg_Of_').reset_index()
res8

Provider Id                                Provider Name  \
0           10001             SOUTHEAST ALABAMA MEDICAL CENTER   
1           10005                MARSHALL MEDICAL CENTER SOUTH   
2           10006               ELIZA COFFEE MEMORIAL HOSPITAL   
3           10007                     MIZELL MEMORIAL HOSPITAL   
4           10008                  CRENSHAW COMMUNITY HOSPITAL   
...           ...                                          ...   
3130       670053                ST LUKE'S SUGAR LAND HOSPITAL   
3131       670054  FOUNDATION SURGICAL HOSPITAL OF SAN ANTONIO   
3132       670055                 METHODIST STONE OAK HOSPITAL   
3133       670059                   ST LUKES LAKESIDE HOSPITAL   
3134       670061                SOUTH TEXAS SURGICAL HOSPITAL   

      Avg_Of_Outpatient Services  Avg_Of_Average  Estimated Submitted Charges  
0                         776.35                                      1942.91  
1                         236.22                                       903.86  
2                         293.59                                      1086.90  
3                          77.00                                      1033.92  
4                          64.75                                      1070.16  
...                          ...                                          ...  
3130                       49.00                                      2843.19  
3131                       32.00                                      3158.36  
3132                       97.67                                      4032.58  
3133                       80.43                                      3234.60  
3134                       57.33                                      4791.18  

[3135 rows x 4 columns]

In [47]:
res8.sort_values(by=['Avg_Of_Outpatient Services','Avg_Of_Average  Estimated Submitted Charges'], ascending=[False,False]).head(10)

Provider Id                                    Provider Name  \
2605       450054                  SCOTT & WHITE MEMORIAL HOSPITAL   
2119       360180                                 CLEVELAND CLINIC   
1352       220171                            LAHEY CLINIC HOSPITAL   
838        140010                                EVANSTON HOSPITAL   
948        140276                 LOYOLA UNIVERSITY MEDICAL CENTER   
2218       370215                          OKLAHOMA HEART HOSPITAL   
1328       220071                   MASSACHUSETTS GENERAL HOSPITAL   
3075       520098  UNIVERSITY OF WI  HOSPITALS & CLINICS AUTHORITY   
1351       220163                UMASS MEMORIAL MEDICAL CENTER INC   
1935       340014                        FORSYTH MEMORIAL HOSPITAL   

      Avg_Of_Outpatient Services  Avg_Of_Average  Estimated Submitted Charges  
2605                    13636.76                                      1212.25  
2119                    13184.00                                      1569.88  
1352                     8824.48                                       855.20  
838                      7844.39                                      1057.43  
948                      6438.03                                       919.96  
2218                     6354.20                                       690.55  
1328                     6319.39                                      1939.97  
3075                     5613.69                                       851.03  
1351                     5537.84                                       923.02  
1935                     5534.24                                       893.32

On the basis of the highest number of average of Outpatient Services and average Estimated Submitted Charges, the 3 Providers that are least like others are amongst the Outpatients:

1. Scott & White Memorial Hospital
2. Cleveland Clinic
3. Lahey Clinic HospitaL

As these providers stand as outliers.

------------------------------------------------------

### Part 2B: Regions Least Like Others

In [32]:
# For Inpatients:

res9 = provider_inpatient.groupby(['Hospital Referral Region (HRR) Description'])[['Total Discharges','Average Covered Charges']].mean().round(2).add_prefix('Avg_Of_').reset_index()
res9

Hospital Referral Region (HRR) Description  Avg_Of_Total Discharges  \
0                               AK - Anchorage                    26.59   
1                              AL - Birmingham                    36.26   
2                                  AL - Dothan                    38.45   
3                              AL - Huntsville                    51.63   
4                                  AL - Mobile                    41.55   
..                                         ...                      ...   
301                                WI - Wausau                    31.47   
302                            WV - Charleston                    45.65   
303                            WV - Huntington                    52.57   
304                            WV - Morgantown                    37.46   
305                                WY - Casper                    25.55   

     Avg_Of_Average Covered Charges  
0                          40348.75  
1                          35875.13  
2                          27610.98  
3                          28310.89  
4                          28951.02  
..                              ...  
301                        24126.26  
302                        19645.20  
303                        23601.86  
304                        19295.39  
305                        31045.17  

[306 rows x 3 columns]

In [34]:
res9.sort_values(by=['Avg_Of_Total Discharges','Avg_Of_Average Covered Charges'], ascending=[False,False]).head(10)

Hospital Referral Region (HRR) Description  Avg_Of_Total Discharges  \
146                             MI - Royal Oak                    85.43   
281                             VA - Lynchburg                    81.29   
51                             DE - Wilmington                    69.68   
177                            NC - Wilmington                    69.36   
153                             MN - St. Cloud                    67.40   
93                               IL - Evanston                    67.28   
193                             NJ - Ridgewood                    66.86   
173                            NC - Greensboro                    66.82   
90                            IL - Blue Island                    65.80   
137                               MI - Detroit                    63.48   

     Avg_Of_Average Covered Charges  
146                        29679.18  
281                        20163.26  
51                         25144.24  
177                        22048.92  
153                        31065.53  
93                         37233.00  
193                        47587.52  
173                        18235.64  
90                         35175.58  
137                        25207.50

On the basis of the highest number of average total discharges and average covered Charges, the 3 Regions that are least like others are amongst the Inpatients:

1. MI - Royal Oak
2. VA - Lynchburg
3. DE - Wilmington

As these regions stand as outliers.

In [33]:
# For Outpatients:

res10 = provider_outpatient.groupby(['Hospital Referral Region (HRR) Description'])[['Outpatient Services','Average  Estimated Submitted Charges']].mean().round(2).add_prefix('Avg_Of_').reset_index()
res10

Hospital Referral Region (HRR) Description  Avg_Of_Outpatient Services  \
0                               AK - Anchorage                      300.76   
1                              AL - Birmingham                      220.81   
2                                  AL - Dothan                      291.68   
3                              AL - Huntsville                      487.28   
4                                  AL - Mobile                      304.24   
..                                         ...                         ...   
299                                WI - Wausau                      834.78   
300                            WV - Charleston                      352.52   
301                            WV - Huntington                      750.81   
302                            WV - Morgantown                      339.51   
303                                WY - Casper                      142.10   

     Avg_Of_Average  Estimated Submitted Charges  
0                                        1272.44  
1                                        1988.59  
2                                        1658.58  
3                                        2226.34  
4                                        1364.44  
..                                           ...  
299                                       913.38  
300                                      1110.63  
301                                      1591.02  
302                                       863.32  
303                                      1498.46  

[304 rows x 3 columns]

In [46]:
res10.sort_values(by=['Avg_Of_Outpatient Services','Avg_Of_Average  Estimated Submitted Charges'], ascending=False).head(10)

Hospital Referral Region (HRR) Description  Avg_Of_Outpatient Services  \
269                                TX - Temple                     8519.43   
179                           ND - Grand Forks                     3650.82   
183                               NH - Lebanon                     3396.38   
25                  CA - Palm Springs/Rancho M                     2333.69   
93                               IL - Evanston                     2167.50   
278                       VA - Charlottesville                     2099.81   
83                             IA - Mason City                     1750.26   
127                                MA - Boston                     1679.99   
177                              ND - Bismarck                     1675.93   
208                             OH - Cleveland                     1656.54   

     Avg_Of_Average  Estimated Submitted Charges  
269                                      1238.90  
179                                       868.57  
183                                       905.73  
25                                       1756.36  
93                                       1551.75  
278                                      1006.06  
83                                        943.64  
127                                      1086.46  
177                                       683.06  
208                                      1345.59

On the basis of the highest number of average of Outpatient Services and average Estimated Submitted Charges, the 3 Regions that are least like others are amongst the Outpatients:

1. TX - Temple	
2. ND - Grand Forks	
3. NH - Lebanon

As these Regions stand as outliers.

-------------